src/core/crypto.js

```diff
@@ -916,20 +916,16 @@ class CipherTransformFactory {
         cipher = new ARCFourCipher(derivedKey);
         checkData = cipher.encryptBlock(checkData);
       }
-      for (j = 0, n = checkData.length; j < n; ++j) {
-        if (userPassword[j] !== checkData[j]) {
-          return null;
-        }
-      }
     } else {
       cipher = new ARCFourCipher(encryptionKey);
       checkData = cipher.encryptBlock(
         CipherTransformFactory._defaultPasswordBytes
       );
-      for (j = 0, n = checkData.length; j < n; ++j) {
-        if (userPassword[j] !== checkData[j]) {
-          return null;
-        }
+    }
+
+    for (j = 0, n = checkData.length; j < n; ++j) {
```

Comment:
nit: can't we use something like `return userPassword.every((x, i) => x === checkData[i]) ? encryptionKey : null;` ?